askama_shared/src/generator.rs

```diff
@@ -593,13 +593,17 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         iter: &Expr<'_>,
         body: &'a [Node<'_>],
         ws2: Ws,
+        else_block: &'a [Node<'_>],
+        ws3: Ws,
     ) -> Result<usize, CompileError> {
         self.handle_ws(ws1);
         self.locals.push();
 
         let expr_code = self.visit_expr_root(iter)?;
 
         let flushed = self.write_buf_writable(buf)?;
+        buf.writeln("{")?;
+        buf.writeln("let mut _did_not_loop = true;")?;
```

Comment:
Let's call this `did_loop` (defaulting to false), which is easier to understand. I don't think it will be unused, since we always read it?

Reply:
Ah, never mind about the _ prefix... we're doing this to prevent clashes with user-defined variables.

Reply:
Renamed to `_did_loop`.